app/javascript/components/profile/contributions-list/BuildingContributionsList.tsx

```diff
@@ -1,25 +1,68 @@
 import React from 'react'
 import { Contribution as ContributionProps } from '../../types'
-import { TrackIcon, Reputation, GraphicalIcon } from '../../common'
+import { TrackIcon, Reputation, GraphicalIcon, Pagination } from '../../common'
 import { fromNow } from '../../../utils/time'
+import { FetchingBoundary } from '../../FetchingBoundary'
+import { ResultsZone } from '../../ResultsZone'
+import { useIsMounted } from 'use-is-mounted'
+import { useList } from '../../../hooks/use-list'
+import { usePaginatedRequestQuery, Request } from '../../../hooks/request-query'
+
+type PaginatedResult = {
+  results: readonly ContributionProps[]
+  meta: {
+    currentPage: number
+    totalCount: number
+    totalPages: number
+  }
+}
+
+const DEFAULT_ERROR = new Error('Unable to load contributions')
```

Comment:
Maybe make this specific?

```suggestion
const DEFAULT_ERROR = new Error('Unable to load building contributions')
```